raiden-ts/src/raiden.ts

```diff
@@ -1387,6 +1397,35 @@ export class Raiden {
     this.store.dispatch(withdraw.request(undefined, meta));
     return promise;
   }
+
+  /**
+   * Fetches an ordered list of suggested partners from provided, configured or first found PFS
+   *
+   * @param token - Token address to get partners for
+   * @param options - Request options
+   * @param options.pfs - PFS to use, instead of configured or automatic
+   * @returns Ordered array of suggested partners, with address and scoring values according to PFS
+   */
+  public async suggestPartners(
+    token: string,
+    options: { pfs?: RaidenPFS } = {},
+  ): Promise<SuggestedPartner[]> {
+    assert(Address.is(token), [ErrorCodes.DTA_INVALID_ADDRESS, { token }], this.log.info);
+    const tokenNetwork = this.state.tokens[token];
+    assert(tokenNetwork, ErrorCodes.RDN_UNKNOWN_TOKEN_NETWORK, this.log.info);
+    const pfs = decode(PFS, options.pfs ?? (await this.findPFS())[0]);
+
+    return await fromFetch(`${pfs.url}/api/v1/${tokenNetwork}/suggest_partner`, {
+      method: 'GET',
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .pipe(
+        timeout(this.config.httpTimeout),
+        mergeMap((response) => response.text()),
+        map((text) => decode(SuggestedPartners, jsonParse(text))),
```

Comment:
What if we have some unexpected decode issues? Let's say there was a version update, the format changed and we are not prepared or... just imagine anything. This would just forward the origin error to the user right? :thinking:

Reply:
Unexpected decode issues will reject the same as network errors, but with its proper error messages. We may include an e2e test for this, once this gets implemented on the dApp. If there's unexpected issues, the best we can do is forward it to the user, they're unexpected afterall.

Reply:
Hmm. Not too happy with it.

Reply:
I understand, but if we'd code for every scenario where an upstream behavior/schema/protocol/api could change, this would be unfeasible.